### PR TITLE
web: core-flow pages on mock data — Home, Chats, Chat, Tasks, Task, Session

### LINKED ISSUE #88: web: core-flow pages on mock data — Home, Chats, Chat, Tasks, Task, Session
## Goal

The core-flow routes — Home `/`, Chat `/chats/:id`, the new Tasks list `/tasks`, Task `/tasks/:id` and Session `/sessions/:id` — render the handoff's structure on mock data using the kit and the fragment, ready for #34 / #40 / #39 to swap the mock reads for actors.

## Requirements

CHT-01, CHT-03, CHT-09, CHT-11, COL-09, COL-12, AGT-09, OPS-07 — see `docs/requirements.md`.

## Design source

`docs/design/HANDOFF.md` → "Screen specs" rows Home, Chat, Add agent dialog, Task, Session; "Content grids"; screenshots `Main.png`, `Chat.png`, `Task.png`, `Session.png` and their artboards.

## Owner paths

- `apps/web/src/pages/{Home,Chat,Chats,Tasks,Task,Session}.tsx`, `apps/web/src/pages/{home,chat,task,session}/**` (sub-components), `apps/web/src/mock/**`, `apps/web/src/components/**`, `apps/web/src/router.ts` (add `/tasks`, `/chats`), `apps/web/src/styles/pages.css` (own sections only), `apps/web/__tests__/pages/**`, `apps/web/__tests__/routes.test.ts`, `apps/web/e2e/core.spec.ts`, `docs/architecture.md` §10 (routes list)

## Depends on

#85, #86, #87

## Scope

- **Mock data** moves to core vocabulary: `TaskStatus`, `WaitReason`, environment descriptors, `AgentCapabilities`, inbox item kinds; one `mock/workspace.ts` shared by all pages with the sample names from the artboards (Atlas, Forge, Lint, Scout; alien01, nuc-lab, platform). `sampleIds` kept for tests.
- **Home**: topbar breadcrumb "Home" + `New chat` primary action. Grid `minmax(0, 1fr) 340px`, gap 28: "Needs you · N open" section (sorted approvals → input → interrupted, oldest first) of `NeedsItem`s with the approval card inline (`ai-approval`), the right rail with "Today · <time zone>" schedule card and "This month" spend card (mono 28/600 + budget bar, the anthropic-api-only note); then the full-width "Active tasks" `DataTable` (`100px 1fr 140px 270px 60px`: status pill + wait reason, objective, assignee tile + name, environment line, age) with `Stop all` danger button opening a `ConfirmDialog` that lists each chain. Empty workspace shows the single "Create your first agent" card.
- **Chats list** `/chats`: same list as the chat page's left column at full width (mobile issue reuses it).
- **Chat**: content flush (padding 0), three columns `280px | 1fr | 320px` separated by `--ag-line` borders. Left: search input + `+` button, chat rows (title, unread badge, member tiles + last line, amber pill when an approval is open). Centre: `Thread` with the "Showing the last 200 entries" chip, messages with tool cards and one open approval, then the `Composer` with the addressing row resolved by the handoff rule (mentions ∩ members, else coordinator, else the single member, else "nobody will answer"). Right: Members (tile, name, role, status pill, environment line, history access line, `Add agent` link), "Tasks in this chat" mini-tree with `Open tree`, `Stop task chain` danger button, and the memory privacy note (MEM-11). Topbar: breadcrumb `Chats › <title>`, search and settings icon buttons. Add-agent dialog (`ConfirmDialog` sizing) asks for history access: "all history" or "from now" (CHT-04). Empty chat = composer only.
- **Tasks list** `/tasks`: the Home tasks table at full width with filter chips by status; breadcrumb parent for Task.
- **Task**: grid `1fr 460px`. Left: delegation tree of `TaskNode`s with the depth counter "depth 1 of 3"; selecting a node swaps the right rail; an open approval for the selected node renders under the tree (`ai-approval`). Right: contract as key-value rows (120 px label column), transitions `Timeline`, result card with `VERIFIED` / `NOT VERIFIED` pill; cancelled chains list children that "could not be stopped" (COL-12).
- **Session**: header tile 44 + id in mono 20 + status pill + environment line. Grid `1fr 400px`. Left: current tool call, compact approval (`compact` mod), event log (mono 12, `44px 110px 1fr` grid, `LIVE` pill, `request` rows amber, `EventsLostRow` fixture). Right: execution card (environment, runtime, account), capabilities list driven by `AgentCapabilities` (supported = primary check, unsupported = text-dim cross + reason; controls for unsupported operations are not rendered at all, AC-15), session grants with `Revoke`.
- `pages.css`: one section per page keyed on `[data-page="home"|"chat"|"tasks"|"task"|"session"]` with the grids above (`repeat(N, minmax(0, 1fr))` for equal columns; fixed rails).
- Every time is rendered in the workspace zone; relative ages switch to a date after 24 h (`components/Age.tsx`).

## Out of scope

- Live actor reads, `connectSession`, posting (#34), approval responses (#40), delegation (#39); responsive rails-drop and mobile (mobile issue); agent and ops pages.

## Acceptance

- [ ] happy-dom route tests: each of the five pages renders its `data-page`, its main regions (by `data-part`/`aria-label`) and the table column template; Home sorts needs-you items in the handoff order; Chat's addressing hint follows the four-case rule; Session hides controls for unsupported operations.
- [ ] e2e `core.spec.ts` at 1280: navigates `/` → chat → task → session via links, asserts the sidebar badge count equals the open needs-you items, and no console errors.
- [ ] Routes test covers `/tasks` and `/chats`; `docs/architecture.md` §10 lists them.

## Verify

```sh
pnpm typecheck && pnpm lint && pnpm test apps/web && pnpm --filter @agentic/web build
pnpm --filter @agentic/web test:e2e
```

## Read first

`docs/design/HANDOFF.md` → Screen specs, Content grids, Edge cases; `packages/ui/src/kit/**`, `packages/ui/src/thread/**`; `docs/architecture.md` §6, §7, §10. Then `AGENTS.md` → "Picking up an issue". Part of the tracking issue #11.

### PR DESCRIPTION
**Pull request by andtii**
Closes #88.

The core-flow routes render the handoff's structure (`docs/design/HANDOFF.md` → Screen specs, Content grids) on a mock workspace in core vocabulary, behind one `load*()` per page so #34 / #40 / #39 swap the reads without touching the markup.

**Pages** (`apps/web/src/pages`)
- **Home `/`**: "Needs you · N open" (approvals → input → interrupted, oldest first inside each kind) as `NeedsItem`s with the `ai-approval` card inline; the rail with "Today · Europe/Stockholm" and "This month" ($18.42 of $50.00, budget bar, the anthropic-api-only note); the active-tasks `DataTable` on `100px 1fr 140px 270px 60px` (status pill + `wait:` line, objective, assignee tile, environment line, age) with "Stop all" opening a `ConfirmDialog` that lists every chain; the workspace empty state when there are no agents. Topbar: "New chat".
- **Chats `/chats`**: the chat list at full width (`pages/chat/ChatList.tsx`, reused by the chat page; the mobile issue reuses it too).
- **Chat `/chats/:id`**: flush three columns `280px | 1fr | 320px` on `--ag-line` borders. List with search, `+`, unread badge / amber pill while an approval is open, member tiles (stack to four, then `+N`). `Thread` with attribution, tool cards with meta (`t_8f2c`, `+18 −6`, `3.4s`), the approval card with its context rows, the STREAMING pill on Lint's row; `Composer` whose "To" row follows the rule mentions ∩ members → coordinator → single member → nobody (`resolveAddressing`, `mentionedIn` reads `@Name` from the draft through the composer's input events). Right: members (tile, role, status, environment line, history access), "Tasks in this chat" mini-tree with "Open tree", "Stop task chain", the MEM-11 note; the add-agent dialog asks "All history" or "From now" (CHT-04). Empty chat = composer only. Breadcrumb `Chats › Mobile pass #47`.
- **Tasks `/tasks`**: the same table for every task with status filter chips (`aria-pressed`); the Task breadcrumb's parent.
- **Task `/tasks/:id`**: the delegation tree of `TaskNode`s with "depth N of 3"; the route's node is selected, clicking another swaps the rail; the open approval for the selected node renders under the tree; right rail `460px` with the contract as key-value rows (120 px label column), transitions `TimelineList`, result with `VERIFIED` / `NOT VERIFIED`; cancelled chains list children that "could not be stopped" (COL-12). Topbar: "Open session", "Stop chain".
- **Session `/sessions/:id`**: header tile 44 + `Session s_41aa` + status + environment line; the current tool call, the compact approval, the event log on `44px 110px 1fr` with `EventsLostRow` and amber `request` rows and the `LIVE` pill; rail `400px` with execution (agent + config version, machine, runtime, account + auth pill, working dir, head, task ref), the capability list from `CapabilityReport` (check / dim cross + reason; controls for unsupported operations are not rendered, AC-15 — "Cancel turn" only when `capabilities.cancel`), session grants with Revoke. The interrupted session (`s3`) carries the named `FailureCard` with Resume.

**Mock** (`apps/web/src/mock/workspace.ts`): agents Atlas / Forge / Lint / Scout with hues and environments, machines alien01 / nuc-lab / platform, tasks as `TaskStatus` + `WaitReason`, the three needs-you items, chats, transcripts built with `createTranscript` (one open permission request bound to the `git push` call), sessions with `CapabilityReport`, today's schedule, spend; `MOCK_NOW` keeps ages stable; `formatAge` switches to a date after 24 h in the workspace zone. `mock/data.ts` keeps every export; its inbox now holds the three kinds that need a person so `needsYouCount()` equals what Home lists.

**Shared** (`apps/web/src/components`): `Page` gains `page` (`data-page` for `styles/pages.css`), `hideTitle` (the `<h1>` stays for assistive tech) and `flush`; `topbar.ts` is a route-keyed registry (`defineTopbar` / `topbarFor`) each page fills with its breadcrumb label and topbar actions, read by `App` from the current route; `Age`, `KeyValue`, `Panel`. `styles/pages.css` holds the per-page grids in the #88 sections plus shared furniture (panels, key-value rows, chips, table cells).

**Acceptance**
- [x] happy-dom route tests: `apps/web/__tests__/pages/{home,chat,task,session}.test.tsx` mount each route through the real route table (`mount.tsx`) and assert `data-page`, the main regions by `aria-label` / `data-part`, and the column template as `<col>` widths; Home sorts needs-you items in the handoff order (`sortNeeds` also tested on a shuffled input); Chat's addressing hint follows the four-case rule (`resolveAddressing` cases + the rendered `addressing` part, and the nobody hint on a chat with two non-coordinators); Session renders no control for unsupported operations and lists them with their reasons. 25 page tests, 38 in `apps/web`.
- [x] e2e `core.spec.ts` at 1280: Home → chat (via "Open chat") → task (via the mini-tree link) → session (via the topbar's "Open session"), the sidebar badge equals the open needs-you items, node selection swaps the rail, no console or page errors; plus no horizontal scroll on seven routes. 6 passed, 2 skipped (phone project, #91's).
- [x] Routes test covers `/tasks` and `/chats` (and that the task crumb leads to `/tasks`); `docs/architecture.md` §10 already lists both since #94.

**Decisions**
- `packages/ui` (additive, outside the owner paths): `describeRequest` on `Thread` / `Message` / `ToolCall` (`DescribeRequestFn` → `ApprovalContext` = the `ApprovalPrompt` props a page knows and the transcript does not: tool name, input, rule, requester, environment, via, decision, compact). Without it the approval card inside a tool call could only show what the reducer has; the handoff's context rows come from the page. CHANGELOG entry added.
- Topbar contributions are a route-keyed registry rather than a signal a page writes on mount: the module is shared across SSR requests in one process, so a page-written signal painted the previous request's actions and breadcrumb (seen on the first screenshots). The registry is a pure function of the route, identical on server and client. The one place a topbar action needs page state (Task's "Stop chain" dialog) uses a module-level signal.
- `App.tsx` (one-line additive changes): renders the `actions` slot from the registry and uses the registry's crumb label. `nav.ts`: `CRUMBS.task` → `/tasks`, `CRUMBS.tasks` added. `apps/web/package.json`: `@sigx/ai-agent` (catalog) and `@agentic/core` added as dependencies — `createTranscript` builds the mock transcripts, core types the view models.
- `e2e/shell.spec.ts` (two lines): the home page's title is "Home", and the badge assertion checks for a positive count rather than the literal `1`, since the mock inbox now holds three items.
- `mock/data.ts` keeps its machine names (`andy-desktop`, `build-box`), so the sidebar strip and the pages disagree on machine names until #46 wires the strip; changing them would break what #89 / #90 read from the same file in parallel.
- Composer recipients follow the draft through the wrapper's `input` events rather than a Composer model, so the fragment stays untouched; the thread does not auto-scroll on first paint (it scrolls on updates), which #34 will see with live data.

**Verify**: `pnpm typecheck && pnpm lint && pnpm test && pnpm size` green (118 files, 1090 passed, 28 skipped; `@agentic/ui` 62.87 kB of 66 kB, design-system CSS 20.16 kB of 22 kB, unchanged by this PR); `pnpm build` green with `zero:validate --strict` at 0 errors, 0 warnings; `pnpm --filter @agentic/web test:e2e` 6 passed, 2 skipped.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: apps/web/src/pages/Task.tsx [hunk 1]
```diff
@@ -1,57 +1,115 @@
...
+                        </p>
+                    ) : null}
+                    {approval ? <ApprovalPrompt request={approval.request} {...approval.context} onRespond={() => undefined} /> : null}
+                </section>
+
+                <aside data-task-rail aria-label="Selected task">
+                    <Panel label="Task contract" slots={{ aside: () => <code data-ref>{contract.ref}</code> }}>
+                        <KeyValue rows={[
+                            { label: 'Objective', value: () => contract.objective },
+                            { label: 'Origin', value: () => <span data-inline-tile><AgentTile name={assignee.name} hue={assignee.hue} size={18} /> {contract.origin}</span> },
```
[the remark this answers is not in the document]
Comment (andtii) [reply apps/web/src/pages/Task.tsx]: Fixed in the follow-up commit.

File: packages/ui/src/thread/ToolCall.tsx [hunk 2]
```diff
@@ -210,8 +217,8 @@ export const ToolCall = component<ToolCallProps>(({ props }) => {
                 {!streaming && sig !== '' && <InputBlock text={inputText(p.input)} />}
                 {output !== undefined && <OutputBlock text={output} logHref={props.logHref} />}
                 {error && <p data-scope={SCOPE} data-part="error">{error}</p>}
-                {awaiting && props.onRespond && <ApprovalPrompt request={request!} onRespond={props.onRespond} toolName={p.name} input={p.input} />}
-                {agent && <AgentCard agent={agent} transcript={props.transcript} onRespond={props.onRespond} onCancelAgent={props.onCancelAgent} />}
+                {awaiting && props.onRespond && <ApprovalPrompt request={request!} onRespond={props.onRespond} toolName={p.name} input={p.input} {...props.describeRequest?.(request!)} />}
```
[the remark this answers is not in the document]
Comment (andtii) [reply packages/ui/src/thread/ToolCall.tsx]: Fixed in the follow-up commit.

File: apps/web/src/pages/Tasks.tsx [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+                    ? (
+                        <DataTable cols={HOME_TASK_COLS} columns={TASK_COLUMNS} label="Tasks">
+                            {rows.map((task) => (
+                                <DataTable.Row>
+                                    <TaskRowCells task={task} />
+                                </DataTable.Row>
+                            ))}
+                        </DataTable>
+                    )
+                    : <EmptyState variant="generic" caption={`No ${st.filter} tasks.`} />}
```
[the remark this answers is not in the document]
Comment (andtii) [reply apps/web/src/pages/Tasks.tsx]: Fixed in the follow-up commit.
